src/query.c

```diff
@@ -980,8 +1017,18 @@ hy_query_filter(HyQuery q, int keyname, int cmp_type, const char *match)
 	HySack sack = query_sack(q);
 	HyReldep reldep = reldep_from_str(sack, match);
 
-	if (reldep == NULL)
-	    return hy_query_filter_empty(q);
+	if (reldep == NULL) {
+        if (cmp_type != HY_GLOB)
+            return hy_query_filter_empty(q);
+        else {
```

Comment:
it should not call `reldep_from_str` in case of HY_GLOB when we know we will not find anything